x/blob/keeper/keeper.go

```diff
@@ -61,9 +61,12 @@ func (k Keeper) PayForBlob(goCtx context.Context, msg *types.MsgPayForBlob) (*ty
 	gasToConsume := uint64(shares.MsgSharesUsed(int(msg.BlobSize)) * GasPerMsgShare)
 	ctx.GasMeter().ConsumeGas(gasToConsume, payForBlobGasDescriptor)
 
-	ctx.EventManager().EmitEvent(
+	err := ctx.EventManager().EmitTypedEvent(
 		types.NewPayForBlobEvent(sdk.AccAddress(msg.Signer).String(), msg.GetBlobSize()),
```

Comment:
```suggestion
		types.NewPayForBlobEvent(msg.Signer, msg.GetBlobSize()),
```
I guess the `msg.Signer` already contains the account address in Bech32 format. What do you think?

Reply:
second this, we should be able to just use the signer. We can test manually by looking at the events in one of the integration tests

Reply:
It doesn't look like that the conversion is as is, I'll confirm via a test

Reply:
@rahulghangas Did you manage to check if it is the same?

Reply:
@SweeXordious `msg.Signer` is the raw form while the `.String()` method encodes it in bech32. Removing the above logic will be breaking change

Reply:
I tried an example and apparently the `msg.Signer` contains:
```
cosmos1n5c5wmz5s2n0d8zpr5ldcl45nkjclaqktls7gd
```

while `sdk.AccAddress(msg.Signer).String()` returns:
```
cosmos1vdhhxmt0wvckudtrx4mk6734wvexuvry8pa8qu34d3jxxmp5x4hxk6nrd3shz6m5d3enwemycqe9zf
```

Which one do we want?

Reply:
After we align on one, should we add a unit test that verifies an event is emitted with the appropriate signer?

Reply:
I'm leaning towards first, since it's more compact and helps us reduce on-chain storage